### ISSUE TITLE
Improve database caching between builds

### ISSUE THREAD

**Issue by nicksantamaria** [integratedexperts/drupal-dev#3]
## Problem

Currently the database dump is downloaded and the SQL file is stored in the `.data` directory which is cached between builds. This is good, however when this file is large (i.e. several gigabytes) the process of importing the db can take a long time.

## Goal

To improve the speed of builds when a large database is in use.

## Proposed solution

The solution will need to support CircleCI 1.0.

* Enable `docker-engine` service on the build image.
* Replace the default local mysql service with one running inside a container.
* After the DB dump has been fetched, create a hash of the file to establish a version.
* If the hash is new, build a new mysql container image with the SQL dump pre-loaded. 
* Tag the new image with the version hash.
* Start the mysql container image.

... build proceeds as usual.

**Comment by nicksantamaria** [comment]
[Build 12](https://circleci.com/gh/nicksantamaria/drupal-dev/12) - first successful build using the containerised DB.

I had to change approach a little bit to keep consistent with my interpretation of the "spirit" of this project.

* The official `mysql` docker image isn't really suitable for this use-case. It _does_ support loading in a schema when the container starts, but there wasn't a reliable way to create a snapshot of the running container _after_ the db had been imported.
* To work around this limitation, I found a more suitable base image - [zanox/mysql](https://hub.docker.com/r/zanox/mysql/)
* Building the container required a Dockerfile, I've created a new project to handle this - [drupal-dev-containers](https://github.com/nicksantamaria/drupal-dev-containers) (if this approach is agreeable, should move the namespace to `integratedexperts`).
* Docker does not support mounting host volumes during `docker build`, meaning the DB dump needed to be fetched by the container during `docker build`. To avoid duplicating the file transfer, the DB dump does not end up on the host machine now.

Also worth noting that I've also discovered that CircleCI 1.0 doesn't support caching system paths, so I will have to find a solution to this.
